File: examples/reticulated_python/11.py
class Fact:
	def mult(self:Mult, x:int, y:int) -> int:
		if x == 0:
			return 0
		else:
			return y+self.mult(x-1, y)
	def fact(self:Mult, x:int) -> any:
		if x == 0:
			return 1
		else:
			return self.mul(x, self.fact(x-1))
	def main(self:Mult) -> dyn:
		return self.mult(6,6)

File: examples/strongscript/11.ts
class Fact {
  mul(x:!number,y:!number):!number { 
     if(x==0){ return 0 } 
     else { y+mul(x+(-1), y) } 
  }
  fact(x:!number):any { 
     if(x==0){ return 1 } 
     else { return mul(x, fact(x+(-1))) } 
  }
  main():any { return this.fact(5) }
}
